_delphi_utils_python/delphi_utils/flash_eval/eval_day.py

```diff
@@ -0,0 +1,415 @@
+
+    export_files = read_filenames(params["common"]["export_dir"])
+    most_recent_d = pd.Series([
+        pd.to_datetime(x.split('_')[0], format="%Y%m%d",errors='raise')
+        if '.git' not in x  else None for (x, y) in export_files ]).dropna().max()
+    source = params["validation"]["common"]["data_source"]
+    file_tup = load_all_files(params["common"]["export_dir"],
+                              most_recent_d-pd.Timedelta('14d'), most_recent_d)
+    signals = params["flash"]["signals"]
+    for signal in signals:
+        curr_df = pd.DataFrame()
+        #start_time = time.time()
+        for date_s in pd.date_range(most_recent_d-pd.Timedelta('14d'),
+                                    most_recent_d-pd.Timedelta('1d')):
+            data = covidcast.signal(source, signal,
+                                    date_s - pd.Timedelta('7d'), date_s,
+                                    geo_type="nation", as_of=date_s)
+            data2 = covidcast.signal(source, signal,
+                                     date_s - pd.Timedelta('7d'), date_s,
+                                     geo_type="state", as_of=date_s)
+            data3 = covidcast.signal(source, signal,
+                                     date_s - pd.Timedelta('7d'), date_s,
+                                     geo_type="county", as_of=date_s)
+            if (data is not None) or ((data2 is not None) or (data3 is not None)):
+                data = pd.concat([data, data2, data3])
+                if data is not None:
+                    data = data[['geo_value', 'value', 'time_value']]
+                    data.columns = ['state', 'value', 'ref']
+                    data['as_of'] = date_s
+                    data['lag'] = (data['as_of'] - data['ref']).dt.days
+                    data = data.set_index(['state', 'lag', 'ref', 'as_of'])
+                    curr_df = pd.concat([data, curr_df])
+        #Add in data gathered today
+        for (filename, _ , data) in file_tup:
+            if signal in filename:
+                as_of = most_recent_d
+                ref = pd.to_datetime(filename.split('_')[0],
+                                     format="%Y%m%d",errors='raise')
+                region = filename.split('_')[1]
+                if region in ['county', 'state', 'nation']:
+                    data = data[['geo_id', 'val']]
+                    data.columns = ['state', 'value']
+                    data['as_of'] = as_of #pd.to_datetime(pd.Timestamp.today())
+                    data['ref'] = ref
+                    data['lag'] = (data['as_of'] - data['ref']).dt.days
+                    data = data.set_index(['state', 'lag', 'ref', 'as_of'])
+                    curr_df = pd.concat([data, curr_df])
+        curr_df = curr_df[~curr_df.index.duplicated(keep='first')].reset_index()
+        #end_time = time.time()
+        #print(f"Total Download Time: {start_time-end_time}")
+
+
+        for lag in range(1,8):
```

Comment:
Is the lag range always fixed? Is max lag something that users can or would want to be able to change in `params`? Even if not, saving max lag (8) as a constant would make this easier to read.